core/container.go

```diff
@@ -495,21 +495,11 @@ func (container *Container) Build(
 	if err != nil {
 		return nil, err
 	}
-	svcs, err := query.Services(ctx)
```

Comment:
Hmmmmmm, now getting very suspicious about the `container.Services.Merge` code above. 

Originally, this was because all the results are evaluated lazily - we don't know where `ExecOp`s might be running - so we have to be able to spin all those up.

But now, we know that when `dockerfileDir`/etc are evaluated, they're "self-contained" - they'll start their own services. I wonder if we can drop the service merging logic entirely??

cc @vito on this one, as the author of service logic, just wanna check my own understanding :sweat_smile:

Reply:
Interesting point. I removed a number of them and tests seem to be happy. I left some merges under the `Publish` and `Export` calls, which are still done via buildkit.

Reply:
I suspect we could *even* remove those. I guess the only reason we would need those is if it was possible to `Publish`/`Export` to a remote `Service`, but there's no dagger-api for that right now.

Even when those `Container`s are built under buildkit, they're still evaluating dagops, which should now be starting their own services.